stile/sys_tests.py

```diff
@@ -744,9 +744,412 @@ def __call__(self, array, linewidth = 0.01, scale = None, figsize = None,
                                 array['g2'] - array['psf_g2'], array['sigma'] - array['psf_sigma'],
                                 linewidth = linewidth, scale = scale,
                                 figsize = figsize, xlabel = r'$x$ [pixel]', ylabel = r'$y$ [pixel]',
-                                size_label = r'$\sigma$ [pixel]', 
+                                size_label = r'$\sigma$ [pixel]',
                                 xlim = xlim, ylim = ylim, equal_axis = True)
 
+###############################################################################
+# Histogram by Song Huang (2014-08-30)
+class HistogramSysTest(SysTest):
+
+    short_name = 'histogram'
+
+    """
+    A base class for Stile systematics tests that generate histograms.
+    """
+
+    """
+    The Scott rule for bin size
+    This function is directly copied from the astroML library
```

Comment:
Going to add another note here that I think we should have these in a separate file with the license at the top, and then import them here, rather than sticking the code in the middle of ours--just to keep things clean.  (I don't remember if these are in a separate file in AstroML--if they are we can just copy the whole file, otherwise we can limit to the bits we want.)